### PR TITLE
feat!: simplified server folder structure (DVOP-620)

### PR DESCRIPTION
**Pull request by khouari1**
[Review Guidance](https://www.notion.so/genesisglobal/Platform-Code-Review-Process-ace93ba760cc4563b0dfb712e2a88d8a)

📓 &nbsp; **Related JIRA**

[PTC-0](https://genesisglobal.atlassian.net/browse/PTC-0)

🤔 &nbsp; **What does this PR do?**

- Add bullets..

🚀 &nbsp; **Where should the reviewer start?**

Add file / directory pointers.

📑 &nbsp; **How should this be tested?**

```
npx -y @genesislcap/genx@latest init prtest -x --ref YOUR-BRANCH-NAME
```

✅ &nbsp; **Checklist**

- [ ] I have tested my changes.
- [ ] I have added tests for my changes.
- [ ] I have updated the project documentation to reflect my changes.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (cistov) [inline .github/pull_request_template.md:1]: Accidental delete?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -2,9 +2,7 @@ name: Build
 
 on:
   push:
-    branches: [main, master, develop, alpha]
   pull_request:
-    branches: [main, master, develop, alpha]
   workflow_dispatch:
```
Comment (cistov) [inline .github/workflows/build.yml:6]: Should be reverted

File: .github/workflows/upgrade.yml
Comment (cistov) [inline .github/workflows/upgrade.yml:1]: Accidental delete too

File: .gitignore [hunk 2]
```diff
@@ -48,19 +51,14 @@ gradle-app.setting
 
 # Cache of project
 .gradletasknamecache
-server/jvm/server/position-deploy/gradle.properties
+build-cache
```
Comment (cistov) [inline .gitignore:54]: Build cache listed twice now

File: server/{{appName}}-app/src/main/genesis/cfg/{{appName}}-processes.xml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+!   Version             : 1.0
+!   Copyright           : (c) Genesis
+!   Date                : 2022-03-18
+!   Function : Provide Process Definitions Config for multi-pro-code-test.
+!
+!   Modification History
+!
+-->
+<processes>
+<!--    <process name="{{upperCase appName}}_DATASERVER">-->
```
Comment (cistov) [inline server/{{appName}}-app/src/main/genesis/cfg/{{appName}}-processes.xml:14]: This is intentionally commented out?

Comment (khouari1) [reply server/{{appName}}-app/src/main/genesis/cfg/{{appName}}-processes.xml:14]: Yes, these are here to help the user get started. They may not apply to all projects so they are commented out.

**Review by cistov — APPROVED** [review]

**Pull request merged**